src/openrct2/CmdlineSprite.cpp

```diff
@@ -452,6 +457,157 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
         sprite_file_close();
         return 1;
     }
+    else if (_strcmpi(argv[0], "exportalldat") == 0)
+    {
+        if (argc < 3)
+        {
+            fprintf(stdout, "usage: sprite exportall <spritefile> <output directory>\n");
+            return -1;
+        }
+
+        char datName[9] = { 0 };
+        std::fill_n(datName, 8, ' ');
+        int32_t i = 0;
+        for (const char* ch = argv[1]; *ch != '\0' && i < 8; ch++)
+        {
+            datName[i++] = *ch;
+        }
+
+        auto context = OpenRCT2::CreateContext();
+        context->Initialise();
+
+        const ObjectRepositoryItem* ori = object_repository_find_object_by_name(datName);
+        if (ori == nullptr)
+        {
+            fprintf(stderr, "Could not find the object.\n");
+            return -1;
+        }
+
+        const rct_object_entry* entry = &ori->ObjectEntry;
+        void* loadedObject = object_manager_load_object(entry);
+        if (loadedObject == nullptr)
+        {
+            fprintf(stderr, "Unable to load object.\n");
+            return -1;
+        }
+        auto entryIndex = object_manager_get_loaded_object_entry_index(loadedObject);
+        uint8_t objectType = entry->GetType();
+
+        auto& objManager = context->GetObjectManager();
+        auto metaObject = objManager.GetLoadedObject(objectType, entryIndex);
+
+        char outputPath[MAX_PATH];
+        safe_strcpy(outputPath, argv[2], MAX_PATH);
+        path_end_with_separator(outputPath, MAX_PATH);
+
+        if (!platform_ensure_directory_exists(outputPath))
+        {
+            fprintf(stderr, "Unable to create directory.\n");
+            return -1;
+        }
+
+        int32_t maxIndex = static_cast<int32_t>(metaObject->GetNumImages());
+        int32_t imagesOffset = 0;
+        switch (objectType)
+        {
+            case OBJECT_TYPE_RIDE:
+            {
+                auto rideEntry = get_ride_entry(entryIndex);
+                imagesOffset = rideEntry->images_offset;
+                break;
+            }
+            case OBJECT_TYPE_SMALL_SCENERY:
+            case OBJECT_TYPE_LARGE_SCENERY:
+            case OBJECT_TYPE_WALLS:
+            case OBJECT_TYPE_BANNERS:
+            case OBJECT_TYPE_PATH_BITS:
+            {
+                auto& objMgr = OpenRCT2::GetContext()->GetObjectManager();
+                auto obj = objMgr.GetLoadedObject(objectType, entryIndex);
```

Comment:
You already have `objManager` above, just reuse it
```suggestion
                auto obj = objManager.GetLoadedObject(objectType, entryIndex);
```